utils/frame/benchmarking-cli/src/writer.rs

```diff
@@ -399,103 +404,100 @@ impl handlebars::HelperDef for JoinHelper {
 
-	fn test_data(pallet: &[u8], benchmark: &[u8], param: BenchmarkParameter, base: u32, slope: u32) -> BenchmarkBatch {
+	use super::*;
+
+	fn test_data(
+		pallet: &[u8],
+		benchmark: &[u8],
+		param: BenchmarkParameter,
+		base: u32,
+		slope: u32,
+	) -> BenchmarkBatch {
 		let mut results = Vec::new();
 		for i in 0 .. 5 {
-			results.push(
-				BenchmarkResults {
-					components: vec![(param, i), (BenchmarkParameter::z, 0)],
-					extrinsic_time: (base + slope * i).into(),
-					storage_root_time: (base + slope * i).into(),
-					reads: (base + slope * i).into(),
-					repeat_reads: 0,
-					writes: (base + slope * i).into(),
-					repeat_writes: 0,
-					proof_size: 0,
-				}
-			)
+			results.push(BenchmarkResults {
+				components: vec![(param, i), (BenchmarkParameter::z, 0)],
+				extrinsic_time: (base + slope * i).into(),
+				storage_root_time: (base + slope * i).into(),
+				reads: (base + slope * i).into(),
+				repeat_reads: 0,
+				writes: (base + slope * i).into(),
+				repeat_writes: 0,
+				proof_size: 0,
+			})
 		}
 
 		return BenchmarkBatch {
 			pallet: [pallet.to_vec(), b"_pallet".to_vec()].concat(),
 			instance: b"instance".to_vec(),
 			benchmark: [benchmark.to_vec(), b"_benchmark".to_vec()].concat(),
 			results,
-		}
+		};
 	}
 
 	fn check_data(benchmark: &BenchmarkData, component: &str, base: u128, slope: u128) {
 		assert_eq!(
 			benchmark.components,
 			vec![
 				Component { name: component.to_string(), is_used: true },
-				Component { name: "z".to_string(), is_used: false},
+				Component { name: "z".to_string(), is_used: false },
 			],
 		);
 		// Weights multiplied by 1,000
 		assert_eq!(benchmark.base_weight, base * 1_000);
 		assert_eq!(
 			benchmark.component_weight,
-			vec![ComponentSlope {
-				name: component.to_string(),
-				slope: slope * 1_000,
-				error: 0,
-			}]
+			vec![ComponentSlope { name: component.to_string(), slope: slope * 1_000, error: 0 }]
 		);
 		// DB Reads/Writes are untouched
 		assert_eq!(benchmark.base_reads, base);
 		assert_eq!(
 			benchmark.component_reads,
-			vec![ComponentSlope {
-				name: component.to_string(),
-				slope,
-				error: 0,
-			}]
+			vec![ComponentSlope { name: component.to_string(), slope, error: 0 }]
 		);
 		assert_eq!(benchmark.base_writes, base);
 		assert_eq!(
 			benchmark.component_writes,
-			vec![ComponentSlope {
-				name: component.to_string(),
-				slope,
-				error: 0,
-			}]
+			vec![ComponentSlope { name: component.to_string(), slope, error: 0 }]
 		);
 	}
 
 	#[test]
 	fn map_results_works() {
-		let mapped_results = map_results(&[
-			test_data(b"first", b"first", BenchmarkParameter::a, 10, 3),
-			test_data(b"first", b"second", BenchmarkParameter::b, 9, 2),
-			test_data(b"second", b"first", BenchmarkParameter::c, 3, 4),
-		], &AnalysisChoice::default()).unwrap();
-
-		let first_benchmark = &mapped_results.get(
-			&("first_pallet".to_string(), "instance".to_string())
-		).unwrap()[0];
+		let mapped_results = map_results(
```

Comment:
Is there a way to tune the insertion of `\n` between `(` and `&[`. A few pages down the reverse change is introduced.

Reply:
This might be due to inconsistent formatting in the original code, but I'm not sure.

Reply:
I don't think that the formatting of the original code matters much to rustfmt. I suspect this is just an artifact of its line-packing heuristics acting up. 

```toml
overflow_delimited_expr = true
```

might improve the situation though.

Reply:
I didn't see an improvement when I tried this.